acme/http_challenge_webserver.go

```diff
@@ -0,0 +1,85 @@
+package acme
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+)
+
+type HttpChallengeWebserver struct {
+	optPort string
+	start   chan net.Listener
+	end     chan error
+}
+
+func NewHttpChallengeWebserver(optPort string) HttpChallengeMethod {
```

Comment:
In this case, httpChallenge can be renamed to HTTPChallenge if we change it to be exported (like I proposed in my remark to https://github.com/xenolf/lego/issues/36) ?